### PR TITLE
Split by input mapping filters

### PR DESCRIPTION
**Pull request by ondrajodas**
Jira - https://keboola.atlassian.net/browse/COM-1047

### CODE REVIEW

File: src/Application.php [hunk 1]
```diff
@@ -320,4 +334,52 @@ private function sortTransformationRows(array $transformationConfig): array
...
+    {
+        if (empty($row['configuration']['input'])) {
+            return true;
+        }
+        $rowInputMappings = (array) array_combine(
+            array_map(fn($v) => $v['destination'], $row['configuration']['input']),
+            $row['configuration']['input']
+        );
+
+        foreach ($savedTransformation->getInputMappingTables() as $savedInputMappingTable) {
```
Comment (pivnicek) [inline src/Application.php:348]: Just going to comment to make sure I understand what is happening here 😄 .

So, it checks the row and looks if there is already an IM with that `destination`.  If there is already one there, then it returns false if there is a difference in the filters being applied.

That seems ok to me.  But what about columns?  what if the 2 have different columns selected?

File: src/Application.php [hunk 2]
```diff
@@ -49,41 +49,55 @@ public function migrateTransformationConfig(array $transformationConfig): array
...
+                $transformationV2 = TransformationV2::createFromConfig($transformationConfig, $row);
+                $transformationsV2[$transformationKey][] = $transformationV2;
+            } else {
+                $transformationV2 = null;
+                foreach ($transformationsV2[$transformationKey] as $key => $savedTransformationV2) {
+                    if ($this->isSuitableTransformation($row, $savedTransformationV2) === true) {
+                        $transformationV2 = $savedTransformationV2;
+                        $transformationsV2[$transformationKey][$key] = $transformationV2;
+                        break;
+                    }
```
Comment (pivnicek) [inline src/Application.php:61]: So, here, if the `$row` is suitable for migration then it is discarded?  it looks like nothing is being used from the `row` here?

File: src/Application.php [hunk 3]
```diff
@@ -49,41 +49,55 @@ public function migrateTransformationConfig(array $transformationConfig): array
...
+                }
+
+                if (is_null($transformationV2)) {
+                    $transformationV2 = TransformationV2::createFromConfig($transformationConfig, $row);
+                    $transformationsV2[$transformationKey][] = $transformationV2;
                 }
             }
 
-            $this->processRow($row, $transformationsV2[$transformationKey]);
+            $this->processRow($row, $transformationV2);
```
Comment (pivnicek) [inline src/Application.php:70]: aha, it is used here.

**Review by pivnicek — APPROVED** [review]
That method is a little difficult to follow 😄 .

But I think I got it in the end.  Have a question about columns though

File: src/Application.php (hunk 1 above)
Comment (ondrajodas) [reply src/Application.php:348]: @pivnicek 

> So, it checks the row and looks if there is already an IM with that destination. If there is already one there, then it returns false if there is a difference in the filters being applied.

exactly

> That seems ok to me. But what about columns? what if the 2 have different columns selected?

if you have 2 tables with different selected columns, we merges them to one IM, see
https://github.com/keboola/app-transformation-migrate/blob/master/src/ValueObjects/TransformationV2.php#L254L280
but for different e.g. `changeSince` we must create a new transformation

Comment (pivnicek) [reply src/Application.php:348]: aha, rozumim.  diky za vysvetleni

**Pull request merged**